### ISSUE TITLE
[no-misused-promises] Have an option do disallow usage of `void` operator for a Promise

### ISSUE THREAD

**Issue by dko-slapdash** [typescript-eslint/typescript-eslint#4016]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-misused-promises": ["error", { ... }]
  }
}
```

```TS
async function someAsyncFunc() {}
...
void someAsyncFunc(); // <-- does not fail; people overuse it
...
someAsyncFunc(); // <-- fails, and this is what we want
```

**Expected Result**

Would be cool if we had an option (`disallowVoidOperator`, maybe even false by default) that doesn't let people silently use `void` operator for a Promise and thus ignore the promise rejections. People tend to overuse it.

**Actual Result**

It doesn't complain. Because the feature is not implemented.

**Versions**

| package                                   | version |
| ----------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `5.0.0` |
| `@typescript-eslint/parser`               | `5.0.0` |
| `TypeScript`                              | `4.4.4` |
| `ESLint`                                  | `8.0.1` |
| `node`                                    | `16.11.1` |
